atom/browser/api/atom_api_top_level_window.cc

```diff
@@ -69,6 +69,17 @@ v8::Local<v8::Value> ToBuffer(v8::Isolate* isolate, void* val, int size) {
     return buffer.ToLocalChecked();
 }
 
+void EmitNextTickTask(TopLevelWindow* emitter, const std::string& name) {
+  emitter->Emit(name);
+}
+
+void EmitNextTick(TopLevelWindow* emitter, const std::string& name) {
+  content::BrowserThread::PostTask(
+      content::BrowserThread::UI, FROM_HERE,
+      base::Bind(&EmitNextTickTask,
+                 emitter, name));
```

Comment:
* Looks like this is nonrepeating; should this be `bind::BindOnce()`?

* Could avoid bit of unnecessary heap action by using `base::StringPiece` rather than `const std::string&` here

* It's not clear to me how the `std::string` objects created on the stack in `OnWindowBlur` and `OnWindowFocus` stay alive through the deferred action to `EmitNextTickStack`? The previous bullet point will fix this, but asking separately for my own edification on base::Bind's internals